sort.php

```diff
@@ -37,4 +37,8 @@ function bubbleSort($sort_array)
     return $sort_array;
 }
 
-echo implode("", bubbleSort(checkForNextLine(file('php://stdin'))));
\ No newline at end of file
+if(isset($argv[1]) && substr($argv[1], 0, 1) !== '-') {
+    echo implode("", bubbleSort(checkForNextLine(file($GLOBALS['argv'][1]))));
+} else {
+    echo implode("", bubbleSort(checkForNextLine(file('php://stdin'))));
+}
```

Comment:
You have lots of duplicate code in if/else compared to how less is different. Please move the duplicated stuff out of if/else.

Reply:
And `file(`?